### PR TITLE
Fix IPv6 missing brackets in httpProxy.ServerOptions

### LINKED ISSUE #5743: IPv6 try to resolve as DNS
**Describe the bug**
I have an K0s cluster with only IPv6 for Public IPs.
Wenn I try to connect with the Cluster I get an error with the TLS.
Error: Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: XXXX. is not in the cert's altnames: DNS
[XXXX:XXXX:XXXX:XXXX:XXXX::XXXX], DNS:localhost, IP Address:127.0.0.1

**To Reproduce**
1. Connect with a Cluster over IPv6

**Expected behavior**
The TLS Cert should be Valid. The IPv6 Address should be checkt kompletly and not only one part of the IPv6

**Environment (please complete the following information):**
- Lens Version: 5.5.4-latest.20220609.2
- OS: Manjaro/Debian
- Installation method (e.g. snap or AppImage in Linux): Pacman/apt

**Logs:**

**Kubeconfig:**

### PR DESCRIPTION
**Pull request by Dominic1DL**
This pull request should fix: #5743
Signed-off-by: dominic_dl <me@dominic.link>

### CODE REVIEW

File: src/main/context-handler/context-handler.ts [hunk 1]
```diff
@@ -157,6 +157,10 @@ export class ContextHandler implements ClusterContextHandler {
       headers.Host = this.clusterUrl.hostname;
     }
 
+    if (headers.Host.includes(":")) {
```
Comment (ixrock) [inline src/main/context-handler/context-handler.ts:160]: So every url with port would use this wrapping?

**Review by ixrock — DISMISSED** [review]
Haven't checked if the fix is really working. 
Otherwise code LGTM.

**Comment by Dominic1DL** [comment]
Hi, I cant find the Question in the comment any more. But in the E-Mail you asked "So every url with port would use this wrapping?" this will not be the case. url.Parse() some lines ahead already splits the port from the Hostname so it will only effect IPv6 addresses, there are the only Host type were ":" could be used. Its not the nicest solution I would say but it works.

**Comment by Nokel81** [comment]
@Dominic1DL It could be nice if there was a comment explaining what exactly this addition does, just to help future people when they are reading the code.

**Comment by ixrock** [comment]
> Hi, I cant find the Question in the comment any more. But in the E-Mail you asked "So every url with port would use this wrapping?" this will not be the case. url.Parse() some lines ahead already splits the port from the Hostname so it will only effect IPv6 addresses, there are the only Host type were ":" could be used. Its not the nicest solution I would say but it works.

Yes, that's what I realized when removed comment after :)

**Review by Nokel81 — DISMISSED** [review]

**Comment by Nokel81** [comment]
@Dominic1DL Please fix lint issues

**Comment by Dominic1DL** [comment]
Hi there, sorry i´m short on Time the last Weeks. There were some Events that needed my attention. I will Provide a fix as soon as possible and I could also Provide a small command explaining the code.

**Comment by Dominic1DL** [comment]
Hi, I managed to find some Time to fix the issues. I also moved my code into the ```if (this.clusterUrl.hostname)``` clause. Becorse I released if this check is false there will be no host param inside headers to check for IPv6.

**Review by Nokel81 — DISMISSED** [review]

**Comment by Nokel81** [comment]
@Dominic1DL Please sign your commits and then we can get this merged

**Comment by Dominic1DL** [comment]
My bad... I forgot to sign it last time... is there an easy way to sign it after the Commit?

**Comment by Nokel81** [comment]
Yes

> To add your Signed-off-by line to every commit in this branch:
>
>    Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
>    In your local branch, run: git rebase HEAD~2 --signoff
>    Force push your changes to overwrite the branch: git push --force-with-lease origin fix_ipv6_cert_altname_err

**Comment by Dominic1DL** [comment]
Thanks done. ;)

**Review by Nokel81 — APPROVED** [review]

**Review by jim-docker — APPROVED** [review]

**Comment by Nokel81** [comment]
Failing tests are unrelated and fixed on master. Merging anyway.

**Pull request merged**